packages/backend/src/controllers/tasks/tasks.controller.ts

```diff
@@ -0,0 +1,295 @@
+    }
+  }
+
+  @Get()
+  @ApiOperation({ summary: 'get the current tasks from a house' })
+  @ApiOkResponse({
+    description: 'tasks retrieved successfully',
+    type: HouseTasksResponseDto,
+  })
+  @ApiBadRequestResponse({ description: 'user is not in a house' })
+  async getTasksForHouse(
+    @User() user: DecodedIdToken,
+  ): Promise<HouseTasksResponseDto | null> {
+    const userDoc = await this.userStoreService.findOneByFirebaseId(user.uid);
+    if (!userDoc.house) {
+      throw new HttpException('user is not in a house', HttpStatus.BAD_REQUEST);
+    }
+
+    const house = await this.houseStoreService.findOne(userDoc.house);
+    if (house) {
+      const tasks = await this.taskStoreService.findAll();
+
+      const tasksForHouse = tasks.filter((task) =>
+        task.house.equals(house._id),
+      );
+
+      const tasksDue = this.taskUtil.checkRecurrence(tasksForHouse);
+      const taskPromises = tasksDue.map((task) =>
+        this.taskStoreService.update(task.id, task.updatedTask),
+      );
+      await Promise.all(taskPromises);
+
+      const updatedTasks = await this.taskStoreService.findAll();
+      const updatedTasksForHouse = updatedTasks.filter((task) =>
+        task.house.equals(house._id),
+      );
+
+      const updatedTasksDto: TaskResponseDto[] = updatedTasksForHouse.map(
+        (task) => {
+          const {
+            name,
+            description,
+            lastCompleted,
+            dueDate,
+            interval,
+            assigned,
+            pool,
+            house,
+          } = task;
+
+          return {
+            name,
+            description,
+            lastCompleted,
+            dueDate,
+            interval,
+            assigned,
+            pool,
+            house,
+            isComplete: task.lastCompleted != null,
+          };
+        },
+      );
+      return {
+        tasks: updatedTasksDto,
+      };
+    }
+  }
+
+  @Delete('/:id')
+  @HttpCode(204)
+  @ApiParam({
+    name: 'id',
+    required: true,
+    description: 'id for the task to delete',
+  })
+  @ApiOperation({ summary: 'delete a task from a house.' })
+  @ApiNoContentResponse({
+    description: 'task successfully deleted.',
+  })
+  @ApiForbiddenResponse({
+    description: 'user is not the owner of the house',
+  })
+  @ApiNotFoundResponse({
+    description: 'task not found',
+  })
+  async deleteTaskFromHouse(@Param('id') id, @User() user: DecodedIdToken) {
+    if (!isValidObjectId(id)) {
+      throw new HttpException('task does not exist', HttpStatus.NOT_FOUND);
+    }
+
+    const task = await this.taskStoreService.findOne(id);
+    if (!task) {
+      throw new HttpException('task not found', HttpStatus.NOT_FOUND);
+    }
+
+    const userDoc = await this.userStoreService.findOneByFirebaseId(user.uid);
+    if (!userDoc.house) {
+      throw new HttpException(
+        'user is not in the house',
+        HttpStatus.BAD_REQUEST,
+      );
+    }
```

Comment:
Logically we can probably skip this check to see if the user is in a house and directly check if they are the owner but it might make sense to check so that we have more descriptive errors on our api

Reply:
After some discussion we have decided to leave this as is. We could just fetch the house by the matching owners firebaseId but this is a specialty call and it a bit too complex for the service. I don't mind doing an extra call and the extra httpExecption informatino checking if the user is in a house is more insightful